sympy/solvers/solvers.py

```diff
@@ -839,41 +878,63 @@ def solve(f, *symbols, **flags):
+
     # checking/recording flags
     ###########################################################################
 
+    # We need to keep track of whether f was passed as a single
+    # expression/equation or as a list of equations (including a list of length
+    # 1). If dict=True or set=True is not passed then the output format from
+    # solve is different depending on whether the inputs were passed as a list
+    # or not.
+    bare_f = not iterable(f)
+
     # set solver types explicitly; as soon as one is False
     # all the rest will be False
-    hints = ('cubics', 'quartics', 'quintics')
-    default = True
-    for k in hints:
-        default = flags.setdefault(k, bool(flags.get(k, default)))
-
-    # allow solution to contain symbol if True:
-    implicit = flags.get('implicit', False)
-
-    # record desire to see warnings
-    warn = flags.get('warn', False)
-
-    # this flag will be needed for quick exits below, so record
-    # now -- but don't record `dict` yet since it might change
-    as_set = flags.get('set', False)
-
-    # keeping track of how f was passed
-    bare_f = not iterable(f)
+    if not cubics:
+        quartics = quintics = False
+    elif not quartics:
+        quintics = False
 
     # check flag usage for particular/quick which should only be used
     # with systems of equations
-    if flags.get('quick', None) is not None:
-        if not flags.get('particular', None):
-            raise ValueError('when using `quick`, `particular` should be True')
-    if flags.get('particular', False) and bare_f:
+    if quick and not particular:
+        raise ValueError('when using `quick`, `particular` should be True')
+    if bare_f and particular:
         raise ValueError(filldedent("""
             The 'particular/quick' flag is usually used with systems of
             equations. Either pass your equation in a list or
             consider using a solver like `diophantine` if you are
             looking for a solution in integers."""))
 
+    # These are the flags that will be passed recursively to _solve,
+    # _solve_system or solve_undetermined_coefficients. In some cases for
+    # example solve_undetermined_coefficients solve will be called again
+    # recursively and these flags need to be passed through here.
+    flags = {
+        # general options
+        'warn':warn, 'simplify':simplify, 'force':force,
+        # options related to checking
+        'check':check, 'numerical':numerical, 'minimal':minimal,
+        # polynomial root handling in _solve
+        'cubics':cubics, 'quartics':quartics, 'quintics':quintics,
+        # options for linear systems for _solve_system
+        'manual':manual, 'particular':particular, 'quick':quick,
+        # random argument passed in test_issue_8755 but not used anywhere else.
+        '_unrad':_unrad,
+        # random argument passed in test_high_order_multivariate but not used
+        # anywhere else:
+        'incomplete':incomplete,
```

Comment:
Maybe these two options should just be removed. They were not previously documented and are only used by a couple of random test functions.

Reply:
I don't think we need to add undocumented options to the documentation, especially ones that are obviously private. It will just make it harder to remove them in the future if we decide to do that.